doc/api/deprecations.md

```diff
@@ -2712,6 +2711,22 @@ Type: Documentation-only.
 
 Prefer [`message.socket`][] over [`message.connection`][].
 
+### DEP0150: Folder index for "type": "module"
+<!-- YAML
+changes:
+  - version: REPLACEME
+    pr-url: https://github.com/nodejs/node/pull/36918
+    description: Runtime deprecation.
+-->
+
+Type: Documentation (supports [`--pending-deprecation`][])
+
+Previously, `index.js` lookup would apply for `require('pkg')` even for packages
+with a `"type": "module"` field in their package.json
```

Comment:
```suggestion
with a `"type": "module"` field in their `package.json`.
```

Reply:
Are you only deprecating the `index.js` resolution? Not also the extensions, so that `"type": "module"` and `"main": "./entry"` (where the file is `entry.js`) also errors?

Reply:
This PR as written deprecates the `"main"` field entirely in `"type": "module"` packages.